composer/core/state.py

```diff
@@ -874,58 +874,68 @@ def get_model_state_dict(self) -> Dict[str, Any]:
         Returns:
             Dict[str, Any]: The state dict for the model.
         """
-        return self.get_model_and_optimizer_state_dict(model_only=True)[0]
+        if version.parse(torch.__version__) > version.parse('2.1.3'):
+            from torch.distributed.checkpoint.state_dict import StateDictOptions, get_model_state_dict
+            if self.fsdp_state_dict_type not in [None, 'full', 'sharded']:
+                raise NotImplementedError(
+                    textwrap.dedent(f'fsdp_state_dict_type={self.fsdp_state_dict_type} is not supported for '
+                                    f'torch version {{version.parse(torch.__version__)}} > 2.1.3. Please set '
+                                    'fsdp_state_dict_type to None, "full", or "sharded".'))
 
-    def _legacy_get_model_state_dict(self) -> Dict[str, Any]:
-        if self.fsdp_enabled and self.fsdp_state_dict_type is not None:
-            with fsdp_state_dict_type_context(self.model, state_dict_type=self.fsdp_state_dict_type):
-                model_state_dict = self.model.state_dict()
+            model_state_dict = get_model_state_dict(
```

Comment:
does this function work for non distributed training? If so...awesome

Reply:
and for ddp does it consume the `module` prefix as we do below?

Reply:
Yes to non distributed, they are unifying interface.

No to consuming. Good catch will add